### ISSUE TITLE
Remove albumRipper.java

### ISSUE THREAD

**Issue by cyian-1756** [RipMeApp/ripme#423]
albumRipper.java is reduntent as everything it does abstractHTMLRipper also does. The rippers that use albumRipper.java should be rewritten to use abstractHTMLRipper.

The rippers that use albumRipper are :

src/main/java/com/rarchives/ripme/ripper/rippers/NewsfilterRipper.java

src/main/java/com/rarchives/ripme/ripper/rippers/TwitterRipper.java

src/main/java/com/rarchives/ripme/ripper/rippers/TumblrRipper.java

src/main/java/com/rarchives/ripme/ripper/rippers/ImgurRipper.java

src/main/java/com/rarchives/ripme/ripper/rippers/MotherlessRipper.java

src/main/java/com/rarchives/ripme/ripper/rippers/PhotobucketRipper.java

~~src/main/java/com/rarchives/ripme/ripper/rippers/VineRipper.java~~

src/main/java/com/rarchives/ripme/ripper/rippers/TeenplanetRipper.java

src/main/java/com/rarchives/ripme/ripper/rippers/RedditRipper.java

src/main/java/com/rarchives/ripme/ripper/rippers/PornhubRipper.java

src/main/java/com/rarchives/ripme/ripper/rippers/SmuttyRipper.java

src/main/java/com/rarchives/ripme/ripper/rippers/VkRipper.java

src/main/java/com/rarchives/ripme/ripper/rippers/NfsfwRipper.java

**Comment by amolrbhagwat** [comment]
Looking at picking up TumblrRipper to rewrite. Is anybody already working on it?

**Comment by cyian-1756** [comment]
@amolrbhagwat

> Looking at picking up TumblrRipper to rewrite. Is anybody already working on it?

As far as I know no one else is working on it

**Comment by aaether32323** [comment]
@cyian-1756 Wouldn't a rewrite of RedditRipper use AbstractJSONRipper instead (or in addition)? I'm sorry if this is dumb question as I'm not fully acquainted with the codebase yet.

**Comment by cyian-1756** [comment]
@aaether32323 

> Wouldn't a rewrite of RedditRipper use AbstractJSONRipper instead

Yup

**Comment by PeterSzakacs** [comment]
Hello there,

I'm currently working on an assignment where I have to pick an open-source project and contribute to it. From the description, I really like your app (well, more the idea of it since I rarely download much stuff, though that could change :wink: ) and this issue seems to fit the assignments parameters (and most of my coding experience has been with Java). 

I've checked the source code and the "API guide" (How to create a site ripper) on the wiki and am currently trying it out, but I have some questions before i fully start: 

- As there has not been much activity recently, what is the current status of this issue, is anyone assigned to it or working on it?
- I have noticed that currently the test of PhotobucketRipper has been completely commented out due to issue #229 , so in effect, there is currently no test, so I was wondering if as part of solving the issue this test should also be reenabled and maybe fixed.

**Comment by cyian-1756** [comment]
@PeterSzakacs 

> As there has not been much activity recently, what is the current status of this issue, is anyone assigned to it or working on it?

This issue is currently open to anyone who wants to work on it, there is no one currently assigned to it

> I have noticed that currently the test of PhotobucketRipper has been completely commented out due to issue #229 , so in effect, there is currently no test, so I was wondering if as part of solving the issue this test should also be reenabled and maybe fixed.

The test failing is a separate issue

**Comment by PeterSzakacs** [comment]
@cyian-1756 

Ok, thanks for the answers, currently working on this issue. :)

**Comment by PeterSzakacs** [comment]
Ok, guys, minor progress update.

Currently I have managed to rewrite the following rippers to inherit from AbstractHTMLRipper. All changes have been commited and pushed to my forks dedicated topic branch:

- src/main/java/com/rarchives/ripme/ripper/rippers/NewsfilterRipper.java
- src/main/java/com/rarchives/ripme/ripper/rippers/NfsfwRipper.java
- src/main/java/com/rarchives/ripme/ripper/rippers/PhotobucketRipper.java
- src/main/java/com/rarchives/ripme/ripper/rippers/PornhubRipper.java
- src/main/java/com/rarchives/ripme/ripper/rippers/TeenplanetRipper.java

More details about the rewrite are in the relevant commit messages (1 commit per ripper). I've also done some basic testing and added or changed some tests as appropriate. Also, SmuttyRipper I have noticed already inherits AbstractHTMLRipper, so there was nothing to do there anyway.

I've also analyzed the source code for all other rippers, and next in line will probably be either VkRipper or TwitterRipper (the latter will use AbstractJSONRipper though).

I'm also writing to ask about further progress. As far as I can see, the individual rippers are independent from each other, and this task itself can be in theory subdivided into as many steps as there are rippers to refactor (plus probably an additional step for in-depth testing). Therefore, I am currently considering whether this issue won't be solved via multiple pull requests addressing only parts of this issue.

Normally, if this were possible to fix one ripper a day, I would not mind first finishing all rippers and only then submitting a PR, but I'm afraid these were the easy ones to refactor. While I have gotten quite confident with the codebase, all further rippers require some major analysis including probably manual API call tracing to figure out how they work currently (particularly Imgur ripper). As this would take a considerable ammount of time and taking into account the speed with which the project is developing (I'm already 12 commits behind upstream/master on my topic branch) as well as the fact that I'll be taking a break to focus on other assignments, I've been wondering if it would be a good idea to submit a pull request now (I can merge the latest commits into my branch locally, test the relevant rippers and push any needed changes back to my fork if needed).
